src/examples/Logic_fastaSearch.py

```diff
@@ -0,0 +1,225 @@
+                            except:
+                                break
+                            counter2 = counter2 + 1
+                        counter = counter + len(stringValue) + len(seqs) + counter2
+                        file_content.seek(counter, 0)
+                        dictKeyAccession[key] = stringValue
+                        proteinList.append(stringValue)
+                        proteinNameList.append(name)
+                        if (os_upper_index == -1):
+                            proteinOSList.append(os)
+                        else:
+                            proteinOSList.append((seqs[descr_upper_index+3:os_upper_index]))
+                    else:
+                        key = seqs.split("OS=")[0]
+                        descr_upper_index = seqs.find('OS=')
+                        # herausfinden bis zu welchem index das OS geht
+                        os_upper_index = seqs.find('(')
+                        os = ''
+                        if (os_upper_index == -1):
+                            zwischenergebnis = seqs[descr_upper_index+3:]
+                            zwischenergebnis1 = zwischenergebnis.split()
+                            os = zwischenergebnis1[0] + ' ' + zwischenergebnis1[1]
+                        name = seqs.split("OS=")[0]  # (seqs[bounds[1]+1:descr_upper_index])
+                        stringValue = ""
+                        counter2 = 0
+                        nextLine = next(file_content)
+                        while not nextLine.startswith('>'):
+                            stringValue += nextLine[:-1]
+                            try:
+                                nextLine = next(file_content)
+                            except:
+                                break
+                            counter2 = counter2 + 1
+                        counter = counter + len(stringValue) + len(seqs) + counter2
+                        file_content.seek(counter, 0)
+                        dictKeyAccession[key] = stringValue
+                        proteinList.append(stringValue)
+                        proteinNameList.append(name)
+                        if (os_upper_index == -1):
+                            proteinOSList.append(os)
+                        else:
+                            proteinOSList.append((seqs[descr_upper_index+3:os_upper_index]))
+        return dictKeyAccession, proteinList, proteinNameList, proteinOSList, dictKeyAccessionDECOY, proteinListDECOY, proteinNameListDECOY, proteinOSListDECOY
+
+    # um im Internet das Protein aufzurufen für mehr Informationen
+
+    def moreProteinInformation(protein_accession):
+        return webbrowser.open("https://www.uniprot.org/uniprot/" + protein_accession)
+
+
+# wird für die Methode protein_dictionary benötigt (suche von mehreren Indizes)
+def find_all_indexes(input_str, search_str):
+    """Gets fasta File Path and saves all infromations about the proteins 
+    in separate lists 
```

Comment:
I think most of the code (above and here) could be shortened significantly by using regular expressions.
See e.g.,
maybe just use https://stackoverflow.com/questions/4664850/how-to-find-all-occurrences-of-a-substring